domain/domain.go

```diff
@@ -817,7 +817,16 @@ func (do *Domain) Init(ddlLease time.Duration, sysExecutorFactory func(*Domain)
 	if err != nil {
 		return err
 	}
-	do.globalCfgSyncer = globalconfigsync.NewGlobalConfigSyncer(do.etcdClient)
+	// try cast do.store to kv.Storage
+	if store, ok := do.store.(interface {
+		GetPDClient() pd.Client
+	}); ok {
+		do.globalCfgSyncer = globalconfigsync.NewGlobalConfigSyncer(store.GetPDClient())
+	} else {
+		logutil.BgLogger().Warn("could not properly initialize GlobalConfigSyncer, due to store does not contain GetPDClient method")
+		do.globalCfgSyncer = globalconfigsync.NewGlobalConfigSyncer(nil)
+	}
```

Comment:
1. you can define the interface in `kv/kv.go`, just after the `Storage` interface.
2. the code can be simply to:
```
	var pdClient pd.Client
	if store,ok := do.store.(kv.StorageWithPD){
		pdClient = store.GetPDClient()
	}
	do.globalCfgSyncer = globalconfigsync.NewGlobalConfigSyncer(pdClient)
```